### ISSUE TITLE
Update qemu version

### ISSUE THREAD

**Issue by grahamwhaley** [kata-containers/runtime#1253]
# Description of problem

According to the [version check tool](https://github.com/kata-containers/runtime/pull/1243) we are way behind on our qemu version (but we know that ;-).
We should spin up a v3.1.x QEMU and give it a testing at least to progress this.

@egernst I'm happy to take a stab at this if nobody else leaps on it.

```
 Num:                                     Item          Current Ver         Upstream Ver
PASSES:
   0:    assets.hypervisor.nemu-ovmf.uscan-url                  0.6                  0.6
   1:           externals.kubernetes.uscan-url            1.13.3-00               1.13.3

FAILURES:
   0:  assets.hypervisor.firecracker.uscan-url               0.12.0               0.14.0
   1:         assets.hypervisor.nemu.uscan-url           2018-11-29           2018-12-17
   2:         assets.hypervisor.qemu.uscan-url                 2.11                3.1.0
   3:                  assets.kernel.uscan-url              4.14.67             4.14.101
   4:         externals.gometalinter.uscan-url                2.0.5                3.0.0
   5:            externals.openshift.uscan-url               3.10.0               3.11.0
   6:                 externals.runc.uscan-url            1.0.0-rc5            1.0.0-rc6
   7:               languages.golang.uscan-url               1.10.4               1.11.5
   8:                      specs.oci.uscan-url            1.0.0-rc5                1.0.1

PASSES: 2
FAILS: 9
```

**Comment by grahamwhaley** [comment]
We should discuss then if we update to a qemu-lite-3.1.0, or a full upstream qemu-3.1.0
First, we can probably just test to see if and how we work with an upstream 3.1.0.
For reference, if we look at our [qemu-lite-2.11.0 branch](https://github.com/kata-containers/qemu/commits/qemu-lite-2.11.0), there are 30 non-merge commits there that are qemu-lite specific. We would need to port and verify each of those:

```
4f33697 piix4: add reset callback for no firmware case
da2efed7 pc: static_prt disabled by default.
8ba3a0f migration: add capability to bypass the shared memory
6ba2bfb piix4: enable pm registers if there is no firmware
c24295f q35: disable smm if there is no firmware
e49700e q35:if there is no fw, no one is going to use PAM, so disable it
5ce8050 q35: initialize MMCFG base when there is no firmware
3ad8a6e q35 acpi: build static _PRT
6ce78e0 vhost-9p: Enabled tag and path configurations
3482bbd vhost-9p: Add support for vhost-9p.
831e8d3 pc: add sanity check
3d2c96b 9p: removing coroutines of 9p to increase the I/O performance
8cfe0af 9p: getattr: use fstat if we have a fd
4193b3f pc.c : support processor which doesn't support 1G page
4de36f7 acpi-build-nofw: don't create overlapped memory region
dbf64da nvdimm: don't preallocate memory for guest nvdimm
c51f1a9 rcu: reduce more than 7MB heap memory by malloc_trim()
8f89b67 pci: option rom is not needed if there is no firmware
ed81292 kvmvapic: disable kvmvapic when hardware tpr acceleration is supported
52e405b pc: If nowfw is set, do not setup PAM
1ce1229 Disabled smm if there is no firmware on 440fx.
0e0ff3d hw/loader: load guest ELF kernel with no host copy for pc-lite
b3b3bfb hw/loader: load ROM via callback function
9ce1259 pc: skip firmware
3c0a4ca pc: support direct loading protected/long mode kernel
e057f7d ich9: enable pm registers when there is no firmware
84b8a11 acpi: patch guest ACPI when there is no firmware
c1a8755 acpi: expose acpi_checksum()
d17c628 acpi: expose data structurs and functions of BIOS linker loader
6a7935b acpi: build static _PRT
```

@kata-containers/architecture-committee for some high level heads up. @egernst who is looking at updates etc. @devimc who looks to have done the updates to qemu-lite-2.11.0

Something historical I don't remember why is that we also have a [qemu-lite-2.11.2 branch](https://github.com/kata-containers/qemu/commits/qemu-lite-2.11.2), which afaik we don't use. Anybody remember why?

**Comment by sboeuf** [comment]
@grahamwhaley 

The commit `6a7935b acpi: build static _PRT` gives us some boot improvement IIRC, but other than that, the main improvements are related to 9p. And with `virtio-fs` coming, I think the goal will be to drop 9p. So I think we can, for now, stick with qemu-lite, and prepare for the Qemu 3.2 or 3.3 where virtio-fs will appear.

There are a few improvement related to `nofw`, but we don't even use it by default. We should stick to pc and q35 without those non-upstream patches, this would relieve us from the Qemu burden.

**Comment by grahamwhaley** [comment]
For reference, I did a [metrics report](https://github.com/kata-containers/tests/tree/master/metrics/report) run of our current qemu-lite-2.11 vs the latest release v3.1.0 (which lists itself as 3.1.50 on the cmdline?).
Attaching the pdf output here.
[metrics_report.pdf](https://github.com/kata-containers/runtime/files/2885838/metrics_report.pdf)

A quick summary then:
 - PSS footprint the same, which was a nice surprise
 - system level scaling, maybe v3.1 is slightly better - not any worse.
 - boot to workload. oh dear. There is like a 39% bump (in the wrong direction) for v3.1 on my machine.
 - Storage for v3.1 looks worse (which I don't really know why that would be - might have to look at our 30 patches in 'lite' to see if there is something relevant there).
 - cpu cycles on network test are effectively unchanged.

Overall then, we would want to look into both boot time and storage numbers I think before we moved to update.
